src/App.js

```diff
@@ -4,25 +4,69 @@ import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
 
 
-const useSemiPersistentState = () =>{
+function App() {
+
+    const listReducer = (state, action) => {
+        switch (action.type) {
+            case 'TODO_LIST_FETCH_INIT':
+                return {
+                    ...state,
+                    isLoading: true,
+                    isError: false,
+                };
+            case 'TODO_LIST_FETCH_SUCCESS':
+                return {
+                    ...state,
+                    isLoading: false,
+                    isError: false,
+                    data: action.payload,
+                };
+            case 'TODO_LIST_FETCH_FAILURE':
+                return {
+                    ...state,
+                    isLoading: false,
+                    isError: true,
+                };
+            default:
+                throw new Error();
+        }
+    };
 
-    const [todoList, setTodoList] = React.useState(
-        JSON.parse(localStorage.getItem('savedTodoList') || '[]' )
+    const [todoList, setTodoList] = React.useReducer(
+        listReducer,
+        {data: [], isLoading: false, isError: false}
     );
 
 
     React.useEffect(() => {
-        localStorage.setItem('savedTodoList', JSON.stringify(todoList));
-    }, [todoList]);
 
-    return [todoList, setTodoList];
-}
+        setTodoList({ type: 'TODO_LIST_FETCH_INIT' });
 
+        const headers = {
+            Authorization: 'Bearer keyFtGqLWD2Fyqz6p'
+        }
 
+        fetch('https://api.airtable.com/v0/appcbHG1yGAqPMKMu/Default', {headers})
+            .then((response) => response.json())
+            .then((result) => {
+                setTodoList({
+                    type: 'TODO_LIST_FETCH_SUCCESS',
+                    payload: result.records,
+                });
+            })
+            .catch(() =>
+                setTodoList({ type: 'TODO_LIST_FETCH_FAILURE' })
+            );
 
-function App() {
+    }, ['https://api.airtable.com/v0/appcbHG1yGAqPMKMu/Default']);
```

Comment:
I may be reading this wrong, but you don't need to add the URL as a dependency since it's a constant. Also, you'll want to use your `.env.local` file to avoid committing your secret API key